Examples/Tests/initial_distribution/analysis_distribution.py

```diff
@@ -0,0 +1,132 @@
+#! /usr/bin/env python
+
+# Copyright 2019-2020 Yinjian Zhao
+#
+# This file is part of WarpX.
+#
+# License: BSD-3-Clause-LBNL
+
+# This script tests initial distributions.
+# 1 denotes gaussian distribution.
+# 2 denotes maxwell-boltzmann distribution.
+# 3 denotes maxwell-juttner distribution.
+# 4 denotes gaussian position distribution.
+# The distribution is obtained through reduced diagnostic ParticleHistogram.
+
+import numpy as np
+import scipy.constants as scc
+import scipy.special as scs
+
+# print tolerance
+tolerance = 0.009
+print('tolerance:', tolerance)
+
+#===============================
+# gaussian and maxwell-boltzmann
+#===============================
+
+# load data
+h1x = np.genfromtxt("h1x.txt")
+h1y = np.genfromtxt("h1y.txt")
+h1z = np.genfromtxt("h1z.txt")
+h2x = np.genfromtxt("h2x.txt")
+h2y = np.genfromtxt("h2y.txt")
+h2z = np.genfromtxt("h2z.txt")
+
+# parameters of bin
+bin_min = -4.0e-2 * scc.c
+bin_max = +4.0e-2 * scc.c
+bin_num = 50
+bin_size = (bin_max-bin_min)/bin_num
+
+# parameters of theory
+u_rms = 0.01
+gamma = np.sqrt(1+u_rms*u_rms)
+v_rms = u_rms / gamma * scc.c
+
+# compute the analytical solution
+f = np.zeros( bin_num )
+for i in range(bin_num):
+    v = (i+0.5)*bin_size + bin_min
+    f[i] = np.exp(-0.5*(v/v_rms)**2)/(v_rms*np.sqrt(2.0*scc.pi))
+
+# normalization
+f = f / np.amax( f )
```

Comment:
Also, why do we need to normalize? As we know all parameters of the beam, we should be able to test the absolute value directly (which is also important to test from the simulation).

Reply:
Well, it is because I used the option `max_to_unity` in the input file, such that the maximum value of the distribution function `f` is normalized to be one, which is nicer to be play with, and we can get a better tolerance, i.e. 0.009, which is closer to 1, thus I need to do the normalization accordingly. Otherwise, after multiplying by the particle weight, the values of `f` would be very huge or tiny.

Reply:
Double precision floats span from 10**-308 to 10**+308, so as long as you are far from this there should not be too big a precision issue.
The reason I am asking is because it is very easy to make a mistake in WarpX when computing the weight (and in the reduced diags), and this is the only place where that could be tested for all these injection methods. Also, that would be a great example of converting weights to charge, that we could point users to.
Do you think we could avoid normalizing the Reduced diags output, and compute the actual distribution (so that the integral over all particles gives the total charge requested by the user)? I can help if you have questions.